src/ifp/src/InitFloorplan.tcl

```diff
@@ -263,4 +269,33 @@ proc microns_to_mfg_grid { microns } {
   }  
 }
 
+proc cut_rows {block placement_blockages} {
+  utl::info "IFP" 6 "Placement blockages found: [llength $placement_blockages]"
+
+  # Gather rows needing to be cut because of placement blockages
+  set rows_to_cut []
+  set row_placement_blockages [dict create]
+  foreach blockage $placement_blockages {
+    if {![$blockage isSoft]} {
+      foreach row [$block getRows] {
+        if {[tap::overlaps $blockage $row 0 0]} {
+          set row_name [$row getName]
+          if {![dict exists $row_placement_blockages $row_name]} {
+            lappend rows_to_cut $row
+          }
+          dict lappend row_placement_blockages $row_name [$blockage getBBox]
+        }
+      }
+    }
+  }
+
+  # cut rows around placement blockages
+  foreach row $rows_to_cut {
+    tap::cut_row $block $row $row_placement_blockages 0 0 0
+  }
+
+  utl::info "IFP" 21 "Cut rows: [llength $rows_to_cut]"
+  utl::info "IFP" 22 "Rows after cut: [llength [$block getRows]]"
```

Comment:
what is the point of printing rows after cut?
ifp already says how many rows there are